### PR TITLE
Use new demosplan-ui color design tokens

### PR DESCRIPTION
**Pull request by ahmad-demos**
**Ticket:** https://yaits.demos-deutschland.de/T30974

In demosplan-ui, several color design tokens are added (https://github.com/demos-europe/demosplan-ui/pull/108). This PR adapts demosplan-core to that change.

:warning: **Want to test it locally?** The already-known procedure applies:
- change package.json dependency to `"@demos-europe/demosplan-ui": "github:demos-europe/demosplan-ui#29b9f039df02011cde7cabd52ab563989b702208",`
- `cd node_modules/@demos-europe/demosplan-ui`
- `yarn; yarn prepack`

After that, new tokens scss files should be available.

### Changes made in this PR

**Include separate color tokens files for palette, data, brand, and ui**

```
@import '@demos-europe/demosplan-ui/tokens/scss/color.palette';
@import '@demos-europe/demosplan-ui/tokens/scss/color.data';
@import '@demos-europe/demosplan-ui/tokens/scss/color.brand';
@import '@demos-europe/demosplan-ui/tokens/scss/color.ui';
```

The color tokens that have been defined in `demosplan/DemosPlanCoreBundle/Resources/client/scss/token` are partly substituted by the new tokens from demosplan-ui, and partly (when not to be considered "usiversal" design tokens) moved to `_settings-color.dplan.scss`.

**Simplify file structure for settings-color**

- Instead of the "color-palette" and "color-assignment" files, there is only one file `_settings-color.project.dplan.scss` (project) and one file `_settings-color.dplan.scss` (core).
- In the `_settings-color.project.dplan.scss` in Project, only what is required for the overrides is imported. eg. I need `$dp-color-red-dark-1`, so I have to import the `color.palette` first. But I do the overrides before the import that contains the defaults, e.g. `$dp-color-brand-highlight`, and only then are the color.brand tokens imported.
- All `Hex Override` variables we have to import before `tokens/scss/color.*` in the Project.

**Adapt utility classes to new ui tokens**

As several new color tokens have been added, the respective utility classes are added, here, as well. Also they are used where appropriate.

### CODE REVIEW

File: client/js/components/document/DpMapSettingsPreview.vue [hunk 1]
```diff
@@ -73,10 +73,10 @@
             :data-cy="Translator.trans(link.label)"
             :href="href(link)"
             class="o-link"
-            :class="{'color--system-confirm':link.done()}">
+            :class="{'color-ui-status-complete-text': link.done()}">
```
Comment (salisdemos) [inline client/js/components/document/DpMapSettingsPreview.vue:76]: As I wrote in https://github.com/demos-europe/demosplan-ui/pull/108 I personally would omit the `ui-`

**Review by salisdemos — APPROVED** [review]
skimmed through. Got a review tour. Looks legit. Lets see how it feels in the wild. 

my Comment is just a personal opinion...

File: client/js/components/document/DpMapSettingsPreview.vue (hunk 1 above)
Comment (spiess-demos) [reply client/js/components/document/DpMapSettingsPreview.vue:76]: Changed in https://github.com/demos-europe/demosplan-core/pull/1326

**Review by salisdemos — APPROVED** [review]
still legit

**Pull request merged**